Add Filter to MPLS views to hide disabled devices (and their corresponding Services/SDPs/SAPs)

peelman opened the pull request:
Minor tweak to MPLS views so that disabled devices don't gum up the works.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [X] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [X] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

peelman commented:
> I think this needs an option to enable showing disabled devices otherwise users don't get to see all data to make decisions.

i'll see if i can add this as a revision.

peelman commented:
> I think this needs an option to enable showing disabled devices otherwise users don't get to see all data to make decisions.

![image](https://github.com/user-attachments/assets/45465d45-1f7f-4f19-b4fc-a8ca5778c902)


Something like this sufficient?

peelman commented:
@laf missed the tag, sorry.

laf commented:
> @laf missed the tag, sorry.

Update the PR with the code and let's see :)

peelman committed:
Add option to MPLS views to hide disabled devices

peelman committed:
Changes as requested...

peelman commented on includes/html/pages/routing/mpls.inc.php:
```diff
@@ -320,6 +340,11 @@
 
     foreach (dbFetchRows('SELECT b.*, s.svc_oid AS svcId FROM `mpls_sdp_binds` AS b LEFT JOIN `mpls_services` AS s ON `b`.`svc_id` = `s`.`svc_id` ORDER BY `sdp_oid`, `svc_oid`') as $sdpbind) {
         $device = device_by_id_cache($sdpbind['device_id']);
+
```
@murrant I've finally managed to squirrel away some extra time to play with this some more.

Feedback welcome

peelman committed:
alignment

peelman committed:
StyleCI Happiness

peelman committed:
StyleCI

peelman committed:
StyleCI

peelman commented:
@laf @murrant Do i need to rebase this against current?  Should still be good to merge i think.

murrant commented:
Thanks for getting back to this.  It still needs some work particularly because db* methods are deprecated.

1. The "disabled" ui is a bit confusing something like "Device Up or All Devices" (not sure I like my suggestion here, if you can come up with something better that would be great)

2. To fix the raw sql queries:
Say we make MplsLsp extend DeviceRelatedModel and add the relationship:
```php
    /**
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo<Vrf, $this>
     */
    public function vrf(): BelongsTo
    {
        return $this->belongsTo(Vrf::class, 'vrf_oid', 'vrf_oid')
            ->whereColumn('vrfs.device_id', 'mpls_lsps.device_id');
    }
```

Now we could do something like this:
```php
$results = MplsLsp::with(['vrf', 'device'])
    ->when($deviceState !== 'disabled', function($q) {
            $q->whereHas('device', fn($query) => $query->where('disabled', 0));
    })
    ->get()
    ->sortBy('device.hostname')
    ->sortBy('mplsLspName');
```

peelman commented:
Yeah...i don't think that's an elephant i have time to eat, at least not for a while.  I just went down the rabbit hole briefly, and i'm on a strugglebus just getting what you had there working (e.g. there is no `hasAccess` on MPLS LSPs today or permissions defined that would cover it (AFAICT).  And the `BelongsTo` doesn't return BelongsTo objects.  And the `whereColumn` can't find `mpls_lsps`.  Again, not insurmountable, but far more of an undertaking than extending what's there to make the UI suck a little less.

peelman committed:
Merge branch 'master' of https://github.com/librenms/librenms into ignore-disabled-devices-in-MPLS-view

# Conflicts:
#	includes/html/pages/routing/mpls.inc.php

peelman committed:
refactor(mpls): convert MPLS routing page from raw SQL to Eloquent

Replace raw SQL queries in the MPLS routing page with Eloquent model
queries for LSPs, paths, SDPs, SDP binds, services, and SAPs. This
improves maintainability and ensures proper access control via the
hasAccess() scope.

Changes:
- Convert all dbFetchRows() calls to Eloquent query builder
- Add vrf() relationship to MplsService model for VRF name lookups
- Update "Disabled" device filter toggle to "Show/Hide Disabled"
- Use model properties directly instead of array access in SAPs view

Also fixes a bug in Timos.php SDP bind discovery where service lookup
used the wrong variable ($svcId callback param instead of parsed
$svc_oid), adds null-safe operators, and corrects hex parsing indices
for technical correctness per the MIB specification.

peelman committed:
apply StyleCI and Rector fixes to routing/mpls page

StyleCI fixes:
- Add space before closure parameters
- Fix indentation in closures (8 spaces to 4)
- Remove trailing whitespace

Rector fixes:
- Add :void return type to closures without return values
- Change is_integer() to is_int()
- Convert sortBy closure to arrow function

peelman committed:
Fix missing quotes per Copilot suggestion

peelman commented:
@murrant hoping we can merge this in for 26.1.
